### DISCUSSION TITLE
Reflexions about the way to construct the theme system

### DISCUSSION THREAD

**Discussion by Pipazoul** [datatlas-erasme/front#99] [General]
@lutangar  We discussed a long time with @OlivPi  about the theming system of Datatlas
The idea would be to easely switch between themes for the end-user

We thought about thoses core features (@OlivPi dont hesitate to correct me ;) )

- The theme name is declared in the instanceconf.json file on the back, the front should be able to switch without a crash
- The front structure should be more modular with theme declared section (like wordpress widget system) you could put those "widget" on the left or the right sections of the theme
   - Filter Side Panel
   - crowdsourcing buttons 
   - On click on point : 
       - Open Side Panel
       - Open Modal
       - Open Popover

For now thoses two themes are quite different the filtering system of "Industries" and "OpenDataAlim" dont have the same ways to display data causing a lot of code redundancy and some hardcoded params

Do you maybe have some hints on how to construct a durable theme system (libs ? ways to construct the component structure ?)

**Comment by lutangar** [comment]
> The theme name is declared in the instanceconf.json file on the back, the front should be able to switch without a crash

The "instance conf" seems to be what's representing what you're calling a "theme" at the moment. 
But this "instance conf" is also holding a `theme` key so it might get confusing...

> Do you maybe have some hints on how to construct a durable theme system (libs ? ways to construct the component structure ?)

The "theming system" is definitely doable wihout any lib as far as I can see, but yeah, component structure might need some work.

But before moving further down this road, the 3 of us (others are welcome as well) need to sit around a table and do a modeling/wording session. 
The expected outcome are __Typescript__ interfaces, types and objects so may we the common "contracts" to further work together.
